tests/sources/test_sharepoint_online.py

```diff
@@ -3515,6 +3516,24 @@ async def test_user_access_control_doc(self, patch_sharepoint_client):
             assert expected_user in access_control
             all(group in access_control for group in expected_groups)
 
+    @pytest.mark.asyncio
+    async def test_user_access_control_doc_with_null_created_date_time(self, patch_sharepoint_client):
+        async with create_spo_source() as source:
+            patch_sharepoint_client.groups_user_transitive_member_of = AsyncIterator([])
+
+            user = {
+                "id": "2",
+                "UserName": "pre_june2018_user",
+                "EMail": "null@spo.com",
+                "createdDateTime": None,
+            }
+
+            user_doc = await source._user_access_control_doc(user)
+
+            assert user_doc["_id"] == user["EMail"]
+            assert "created_at" in user_doc
+            assert user_doc["created_at"] is None
```

Comment:
We could probably just keep the second assertion here, as it implies the key is also present.

Reply:
Addressed in [07fbbce](https://github.com/elastic/connectors/pull/3542/commits/07fbbced4d39e960fa045351d97be6a9c5c9b4f5)